File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int[] iList = new int[1000001];
        
        iList[0] = 0;
        iList[1] = 1;
        
        for (int i = 2; i < n+1; i++) {
            iList[i] = (iList[i-1] + iList[i-2]) % 1234567;
        }
        
        return iList[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    
    # DP 문제로 풀이해야함
    # (n+2) = n + (n+1)
    
    li = [0, 1]
    
    for i in range(2, n+1):
        li.append((li[i-1] + li[i-2]) % 1234567)
    
    return li[n]